### PR TITLE
Replace styled-components with bulma helpers

### PR DESCRIPTION
**Pull request by fscholdei**
## Proposed changes

Use [Bulma helpers](https://bulma.io/documentation/helpers/) whenever possible instead of custom styled components.
This pull request replaces primarily color definitions, spacing and flex instructions.

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [ ] New code is covered with unit tests
- [x] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Feature has been tested with different permissions
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/ui-webapp/src/containers/OmniSearch.tsx [hunk 1]
```diff
@@ -115,7 +103,7 @@ const AvatarSection: FC<HitProps> = ({ hit }) => {
 const MoreResults: FC<GotoProps> = ({ gotoDetailSearch }) => {
   const [t] = useTranslation("commons");
   return (
-    <ResultFooter className="dropdown-item has-text-centered">
+    <ResultFooter className={classNames("dropdown-item", "has-text-centered", "mx-2", "px-2", "py-1")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/containers/OmniSearch.tsx:106]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/containers/OmniSearch.tsx [hunk 2]
```diff
@@ -156,7 +144,18 @@ const Hits: FC<HitsProps> = ({ showHelp, gotoDetailSearch, ...rest }) => {
   return (
     <>
       <div aria-expanded="true" role="listbox" className="dropdown-content">
-        <ResultHeading className="dropdown-item">
+        <ResultHeading
+          className={classNames(
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/containers/OmniSearch.tsx:148]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/containers/OmniSearch.tsx [hunk 3]
```diff
@@ -331,7 +330,7 @@ const OmniSearch: FC = () => {
   const { onKeyDown, index } = useKeyBoardNavigation(gotoDetailSearch, clearQuery, data?._embedded.hits);
 
   return (
-    <Field className="navbar-item field">
+    <div className={classNames("navbar-item", "field", "mb-0")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/containers/OmniSearch.tsx:333]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/containers/PluginLoader.tsx [hunk 4]
```diff
@@ -129,10 +118,17 @@ class PluginLoader extends React.Component<Props, State> {
         <section className="section">
           <div className="container">
             <ErrorBoundary>
-              <Centered>
+              <div
+                className={classNames(
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/containers/PluginLoader.tsx:122]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/groups/components/table/GroupMember.tsx [hunk 5]
```diff
@@ -59,6 +57,10 @@ export default class GroupMember extends React.Component<Props> {
   render() {
     const { member } = this.props;
     const to = `/user/${member.name}`;
-    return <StyledMember>{this.showName(to, member)}</StyledMember>;
+    return (
+      <StyledMember className={classNames("is-inline-block", "mr-1", "px-3", "py-1")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/groups/components/table/GroupMember.tsx:61]: Is `is-inline-block` an actual bulma class ? I couldnt find it in the docs.

Comment (Pilopa) [reply scm-ui/ui-webapp/src/groups/components/table/GroupMember.tsx:61]: And you dont need `classNames` here.

File: scm-ui/ui-webapp/src/permissions/components/PermissionsWrapper.tsx [hunk 6]
```diff
@@ -56,7 +54,7 @@ export default class PermissionsWrapper extends React.Component<Props> {
     const permissionArray = Object.keys(permissions);
     return (
       <div className="columns">
-        <StyledWrapper className={classNames("column", "is-half")}>
+        <StyledWrapper className={classNames("column", "is-half", "pb-0")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/permissions/components/PermissionsWrapper.tsx:57]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/branches/components/BranchDetail.tsx [hunk 7]
```diff
@@ -22,49 +22,34 @@
...
     return (
       <div className="media">
-        <FlexRow className="media-content subtitle">
-          <Label>{t("branch.name")}</Label> {branch.name} <DefaultBranchTag defaultBranch={branch.defaultBranch} />
-          <Created className="is-ellipsis-overflow">
-            {t("branches.overview.lastCommit")} <Date date={branch.lastCommitDate} className="has-text-grey" />
-          </Created>
-        </FlexRow>
+        <div
+          className={classNames("media-content", "subtitle", "is-flex", "is-align-items-center", "is-flex-wrap-wrap")}
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/branches/components/BranchDetail.tsx:44]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/branches/components/BranchDetail.tsx [hunk 8]
```diff
@@ -22,49 +22,34 @@
...
+        >
+          <strong className="mr-1">{t("branch.name")}</strong> {branch.name}{" "}
+          <DefaultBranchTag defaultBranch={branch.defaultBranch} />
+          <div className={classNames("is-ellipsis-overflow", "is-size-7", "ml-2")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/branches/components/BranchDetail.tsx:48]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/branches/components/BranchRow.tsx [hunk 9]
```diff
@@ -63,9 +58,9 @@ const BranchRow: FC<Props> = ({ baseUrl, branch, onDelete }) => {
           <DefaultBranchTag defaultBranch={branch.defaultBranch} />
         </ReactLink>
         {branch.lastCommitDate && (
-          <Created className="has-text-grey is-ellipsis-overflow">
+          <span className={classNames("has-text-grey", "is-ellipsis-overflow", "is-size-7", "ml-4")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/branches/components/BranchRow.tsx:61]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/branches/components/DefaultBranchTag.tsx [hunk 10]
```diff
@@ -30,17 +30,16 @@ type Props = WithTranslation & {
   defaultBranch?: boolean;
 };
 
-const LeftMarginTag = styled(Tag)`
+const InheritVerticalAlignTag = styled(Tag)`
   vertical-align: inherit;
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/branches/components/DefaultBranchTag.tsx:34]: Why did you remove the vertical-align in a different component but not here ? The default value is `baseline`, does it differ here ?

File: scm-ui/ui-webapp/src/repos/codeSection/containers/FileSearch.tsx [hunk 11]
```diff
@@ -113,8 +101,18 @@ const FileSearch: FC<Props> = ({ repository, baseUrl, branches, selectedBranch }
         switchViewLink={evaluateSwitchViewLink}
       />
       <div className="panel">
-        <InputContainer>
-          <HomeLink to={contentBaseUrl}>
+        <div
+          className={classNames(
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/codeSection/containers/FileSearch.tsx:105]: You dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/components/changesets/ChangesetDetails.tsx [hunk 12]
```diff
@@ -126,21 +90,21 @@ const Contributors: FC<{ changeset: Changeset }> = ({ changeset }) => {
 
   if (open) {
     return (
-      <ContributorDetails>
-        <FlexRow>
-          <ContributorToggleLine onClick={e => setOpen(!open)} className="is-ellipsis-overflow">
+      <div className="is-flex is-flex-direction-column mb-4">
+        <div className="is-flex is-flex-direction-row">
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/components/changesets/ChangesetDetails.tsx:94]: Does it need the flex row class ?

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryForm.tsx [hunk 13]
```diff
@@ -22,28 +22,18 @@
...
+import styled from "styled-components";
 import { ExtensionPoint } from "@scm-manager/ui-extensions";
 import { IndexResources, Repository, RepositoryType, CUSTOM_NAMESPACE_STRATEGY } from "@scm-manager/ui-types";
 import { Checkbox, Level, Select, SubmitButton } from "@scm-manager/ui-components";
 import NamespaceAndNameFields from "../NamespaceAndNameFields";
 import RepositoryInformationForm from "../RepositoryInformationForm";
 
-const CheckboxWrapper = styled.div`
-  margin-top: 2em;
+const FlexibleDiv = styled.div`
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/components/form/RepositoryForm.tsx:33]: You can set flex-grow and flex-shrink with bulma classes.

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx [hunk 14]
```diff
@@ -21,22 +21,12 @@
...
 import { useTranslation } from "react-i18next";
-
-const MarginIcon = styled(Icon)`
-  padding-right: 0.5rem;
-`;
-
-const SmallButton = styled(Button)`
-  border-radius: 4px;
-  font-size: 1rem;
-  font-weight: 600;
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx:38]: What happened with this styling ? Do we need `has-text-weight-semibold` ?

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx [hunk 15]
```diff
@@ -62,10 +52,14 @@ const RepositoryFormButton: FC<RepositoryForm> = ({ path, icon, label }) => {
...
-      <MarginIcon name={icon} color={isSelected ? "white" : "default"} />
-      <p className="is-hidden-mobile is-hidden-tablet-only">{t(`plugins:${label}`, label)}</p>
-    </SmallButton>
+    <Button
+      className="is-size-6"
+      color={isSelected ? "link is-selected" : undefined}
+      link={!isSelected ? href : undefined}
+    >
+      <Icon className="pr-2" name={icon} color={isSelected ? "white" : "default"} />
+      <p className={classNames("is-hidden-mobile", "is-hidden-tablet-only")}>{t(`plugins:${label}`, label)}</p>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx:61]: We dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/containers/ExportRepository.tsx [hunk 16]
```diff
@@ -60,7 +59,7 @@ const ExportInterruptedNotification = () => {
 const ExportInfoBox: FC<{ exportInfo: ExportInfo }> = ({ exportInfo }) => {
   const [t] = useTranslation("repos");
   return (
-    <InfoBox>
+    <InfoBox className={classNames("my-4", "p-4")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/containers/ExportRepository.tsx:62]: We dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/containers/ExportRepository.tsx [hunk 17]
```diff
@@ -144,7 +143,7 @@ const ExportRepository: FC<Props> = ({ repository }) => {
         helpText={t("export.encrypt.helpText")}
       />
       {encrypt && (
-        <div className="columns column is-half">
+        <div className={classNames("columns", "column", "is-half")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/containers/ExportRepository.tsx:146]: We dont need `classNames` here.

File: scm-ui/ui-webapp/src/repos/tags/components/TagDetail.tsx [hunk 18]
```diff
@@ -21,50 +21,34 @@
...
-`;
-
 const TagDetail: FC<Props> = ({ tag, repository }) => {
   const [t] = useTranslation("repos");
 
   return (
     <div className="media">
-      <FlexRow className="media-content">
-        <Label className="subtitle has-text-weight-bold has-text-black">{t("tag.name") + ": "} </Label> <span className="subtitle">{tag.name}</span>
+      <div className={classNames("media-content", "is-flex", "is-flex-wrap-wrap", "is-align-items-center")}>
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/repos/tags/components/TagDetail.tsx:41]: We dont need `classNames` here or anywhere else in this file.

File: scm-ui/ui-webapp/src/users/components/UserConverter.tsx [hunk 19]
```diff
@@ -22,24 +22,18 @@
...
 } from "@scm-manager/ui-components";
-import { useTranslation } from "react-i18next";
-import { User } from "@scm-manager/ui-types";
-import styled from "styled-components";
-import { useConvertToExternal, useConvertToInternal } from "@scm-manager/ui-api";
-
-const ExternalDescription = styled.div`
-  display: flex;
-  align-items: center;
-  font-weight: 400;
```
Comment (Pilopa) [inline scm-ui/ui-webapp/src/users/components/UserConverter.tsx:41]: Why did this have a 400 font-weight ? Was it removed on purpose ?

**Review by Pilopa — CHANGES_REQUESTED** [review]

File: scm-ui/ui-components/src/Breadcrumb.stories.tsx [hunk 20]
```diff
@@ -69,4 +74,17 @@ storiesOf("BreadCrumb", module)
       revision={"1"}
       permalink={"/" + longPath}
     />
+  ))
+  .add("With prefix button", () => (
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-components/src/Breadcrumb.stories.tsx:78]: An error occurred in a test scenario, which can be tested with this in future.

File: scm-ui/ui-components/src/layout/GroupEntry.tsx [hunk 21]
```diff
@@ -72,18 +62,13 @@ const Description = styled.p`
...
 const ContentLeft = styled.div`
-  display: flex;
   flex: 1 1 auto;
-  align-items: center;
   min-width: 0;
 `;
 
 const ContentRight = styled.div`
-  display: flex;
   flex: 0 0 auto;
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-components/src/layout/GroupEntry.tsx:70]: As far as I understood correctly, the grow default is 0 and the shrink default is 1, so I just added `is-flex-shrink-0`.

File: scm-ui/ui-components/src/layout/GroupEntry.tsx [hunk 22]
```diff
@@ -92,10 +77,6 @@ const NameDescriptionWrapper = styled.div`
   flex: 1 1 auto;
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-components/src/layout/GroupEntry.tsx:77]: As far as I understood correctly, the grow default is 0 and the shrink default is 1, so I just added `is-flex-grow-1`.

File: scm-ui/ui-components/src/repos/changesets/ChangesetRow.tsx [hunk 23]
```diff
@@ -130,36 +106,33 @@ class ChangesetRow extends React.Component<Props> {
                     <p className="is-hidden-desktop">
                       <Trans i18nKey="repos:changeset.shortSummary" components={[changesetId, dateFromNow]} />
                     </p>
-                    <FlexRow>
-                      <AuthorWrapper className="is-size-7 is-ellipsis-overflow">
+                    <div className={classNames("is-flex", "is-flex-direction-row")}>
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-components/src/repos/changesets/ChangesetRow.tsx:109]: That's right, I blindly adopted the old definition here.

File: scm-ui/ui-styles/src/scm.scss [hunk 24]
```diff
@@ -826,9 +825,6 @@ form .field:not(.is-grouped) {
     margin-bottom: 0;
   }
 }
-.menu div:first-child .menu-label {
-  background-color: $blue;
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-styles/src/scm.scss:830]: There used to be a main page menu and an action page menu. Since the latter has been completely abolished, you don't need to distinguish the label background-color anymore. Moreover, we would trigger this within the component and not in the main styles.

File: scm-ui/ui-webapp/src/components/LoginForm.tsx [hunk 25]
```diff
@@ -50,7 +50,7 @@ const AvatarImage = styled(Image)`
   width: 128px;
   height: 128px;
   padding: 5px;
-  background: #fff;
+  background: white;
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-webapp/src/components/LoginForm.tsx:53]: Unfortunately, we cannot access the css custom properties because they are not supported by all browsers. white, however, corresponds to the value of $white.

File: scm-ui/ui-webapp/src/groups/components/table/GroupMember.tsx (hunk 5 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/groups/components/table/GroupMember.tsx:61]: This is documented with the [visibility/responsive helpers](https://bulma.io/documentation/helpers/visibility-helpers/).

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx (hunk 14 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx:38]: 600 is still normal thickness.

File: scm-ui/ui-webapp/src/users/components/UserConverter.tsx (hunk 19 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/users/components/UserConverter.tsx:41]: `400` is still normal thickness.

File: scm-ui/ui-webapp/src/containers/Login.tsx [hunk 26]
```diff
@@ -62,15 +57,15 @@ const Login: FC = () => {
   }
 
   return (
-    <HeroSection className="hero">
+    <section className="hero pt-6">
```
[the remark this answers is not in the document]
Comment (fscholdei) [reply scm-ui/ui-webapp/src/containers/Login.tsx:60]: I didn't question every existing styling. But it seems that the [Hero layout](https://bulma.io/documentation/layout/hero/) component was used here. Does that pose a problem from your point of view?

File: scm-ui/ui-webapp/src/repos/branches/components/DefaultBranchTag.tsx (hunk 10 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/branches/components/DefaultBranchTag.tsx:34]: Good hint. From my point of view, this also makes little sense.

File: scm-ui/ui-webapp/src/repos/components/changesets/ChangesetDetails.tsx (hunk 12 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/components/changesets/ChangesetDetails.tsx:94]: No, this is by default.

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryForm.tsx (hunk 13 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/components/form/RepositoryForm.tsx:33]: During the conversion I noticed that the whole construct is not responsive, so I replaced it with .column.

File: scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx (hunk 14 above)
Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/components/form/RepositoryFormSwitcher.tsx:38]: A button has these properties by default.
